XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -215,6 +220,243 @@ private bool FilterIsOn
+                    _nameCell.Value = pool;
+                    _versionCell.Value = master.ProductVersionTextShort;
+                    _requiredUpdateCell.Value = "";
+                    _installedUpdateCell.Value = "";
+                    bool poolPatchingStatus = false;
+
+                    // pool patching status is false only if all its hosts' patching status are false
+                    foreach (Host h in pool.Connection.Cache.Hosts)
+                    {
+                        var hostRequired = RecommendedPatchesForHost(h);
+                        if (hostRequired.Length > 0)
+                        {
+                            poolPatchingStatus = true;
+                            break;
+                        }
+                    }
+
+                    _patchingStatusCell.Value = poolPatchingStatus
+                        ? Properties.Resources._000_error_h32bit_16
+                        : Properties.Resources._000_Tick_h32bit_16;
+                }
+                
+                else if (Tag is Host)
+                {
+                    Host host = (Host)Tag;
+                    var hostRequired = RecommendedPatchesForHost(host);
+                    var hostInstalledList = Helpers.HostAppliedPatchesList(host);
+                    var hostInstalled = string.Join(", ", hostInstalledList.ToArray());
+                    var hostStatus = (hostRequired.Length > 0);
+
+                    if (_hasPool)    // host in pool
+                        _poolIconCell.Value = "";
+                    else     // standalone host
+                        _poolIconCell.Value = Images.GetImage16For(host);
+
+                    _nameCell.Value = host;
+                    _versionCell.Value = host.ProductVersionTextShort;
+                    _patchingStatusCell.Value = hostStatus ? Properties.Resources._000_error_h32bit_16 : Properties.Resources._000_Tick_h32bit_16;
+                    _requiredUpdateCell.Value = hostRequired;
+                    _installedUpdateCell.Value = hostInstalled;
+                }
+            }
+
+            private string RecommendedPatchesForHost(Host host)
+            {
+                var result = new List<string>();
+                var recommendedPatches = Updates.RecommendedPatchesForHost(host);
+
+                foreach (var patch in recommendedPatches)
+                    result.Add(patch.Name);
+
+                return string.Join(", ", result.ToArray());
+            }
+        }
+
+        private void RebuildHostView()
+        {
+            Program.AssertOnEventThread();
+
+            if (!Visible)
+                return;
+
+            if (checksQueue > 0)
+                return;
+
+            SetFilterLabel();
+
+            try
+            {
+                dataGridViewHosts.SuspendLayout();
+
+                if (dataGridViewHosts.RowCount > 0)
+                {
+                    StoreSelectedUpdates();

```

Comment:
It works ok when clicking on the Name column, however clicking on the other columns still mixes hosts and pools.
Also, I still see the vertical truncation when resizing the name column.